runtime/eventlog.c

```diff
@@ -0,0 +1,396 @@
+    (int64_t)t.tv_nsec;
+
+
+#endif
+}
+
+static void setup_evbuf()
+{
+  CAMLassert(!evbuf);
+  evbuf = caml_stat_alloc_noexc(sizeof(*evbuf));
+
+  if (evbuf == NULL)
+    caml_fatal_error("eventlog: could not allocate event buffer");
+
+  evbuf->ev_generated = 0;
+}
+
+#define OUTPUT_FILE_LEN 4096
+static void setup_eventlog_file()
+{
+  char_os output_file[OUTPUT_FILE_LEN];
+  char_os *eventlog_filename = NULL;
+
+  eventlog_filename = caml_secure_getenv(T("OCAML_EVENTLOG_FILE"));
+
+  if (eventlog_filename) {
+    int ret = snprintf_os(output_file, OUTPUT_FILE_LEN, T("%s.%d.eventlog"),
+                         eventlog_filename, Caml_state->eventlog_startup_pid);
+    if (ret > OUTPUT_FILE_LEN)
+      caml_fatal_error("eventlog: specified OCAML_EVENTLOG_FILE is too long");
+  } else {
+    snprintf_os(output_file, OUTPUT_FILE_LEN, T("caml-eventlog-%d"),
+               Caml_state->eventlog_startup_pid);
+  }
+
+  Caml_state->eventlog_out = fopen_os(output_file, T("wb"));
+
+  if (Caml_state->eventlog_out) {
+    int ret =  fwrite(&header, sizeof(struct ctf_stream_header),
+                      1, Caml_state->eventlog_out);
+    if (ret != 1)
+      caml_eventlog_disable();
+    fflush(Caml_state->eventlog_out);
+  } else {
+    caml_fatal_error("eventlog: could not open trace for writing");
+  }
+}
+#undef OUTPUT_FILE_LEN
+
+#define FWRITE_EV(item, size) \
+  if (fwrite(item, size, 1, out) != 1) \
+    goto fwrite_failure;
+
+static void flush_events(FILE* out, struct event_buffer* eb)
+{
+  uintnat i;
+  uint64_t flush_duration;
+  uintnat n = eb->ev_generated;
+
+  struct ctf_event_header ev_flush;
+  ev_flush.id = EV_FLUSH;
+  ev_flush.timestamp = time_counter() -
+                        Caml_state->eventlog_startup_timestamp;
+  ev_flush.pid = Caml_state->eventlog_startup_pid;
+
+  for (i = 0; i < n; i++) {
+    struct event ev = eb->events[i];
+    ev.header.pid = Caml_state->eventlog_startup_pid;
+
+    FWRITE_EV(&ev.header, sizeof(struct ctf_event_header));
+
+    switch (ev.header.id)
+    {
+    case EV_ENTRY:
+      FWRITE_EV(&ev.phase, sizeof(uint16_t));
+      break;
+    case EV_EXIT:
+      FWRITE_EV(&ev.phase, sizeof(uint16_t));
+      break;
+    case EV_COUNTER:
+      FWRITE_EV(&ev.count, sizeof(uint64_t));
+      FWRITE_EV(&ev.counter_kind, sizeof(uint16_t));
+      break;
+    case EV_ALLOC:
+      FWRITE_EV(&ev.count, sizeof(uint64_t));
+      FWRITE_EV(&ev.alloc_bucket, sizeof(uint8_t));
+      break;
+    default:
+      break;
+    }
+  }
+
+  flush_duration =
+    (time_counter() - Caml_state->eventlog_startup_timestamp) -
+    ev_flush.timestamp;
+
+  FWRITE_EV(&ev_flush, sizeof(struct ctf_event_header));
+  FWRITE_EV(&flush_duration, sizeof(uint64_t));
+
+  return;
+
+ fwrite_failure:
+  /* on event flush failure, shut down eventlog. */
+  if (caml_runtime_warnings_active())
+    fprintf(stderr,
+           "eventlog: error while writing trace file, disabling eventlog\n");
```

Comment:
```suggestion
           "[ocaml] error while writing trace file, disabling eventlog\n");
```
If you don't mention ocaml in the message, a user waste a lot of time tracking down the origin of the message. I suggest using the same prefix as all other runtime messages.

Reply:
Sure, I applied the suggestion, thank you.
I think this is the only runtime message added that required this change.